pkg/infrastructure/openstack/postprovision/floatingips.go

```diff
@@ -33,12 +33,12 @@ func FloatingIPs(ctx context.Context, c client.Client, cluster *capo.OpenStackCl
 		return err
 	}
 
-	bootstrapPort, err := getPortForInstance(networkClient, *bootstrapMachine.Spec.InstanceID)
+	bootstrapPort, err := getPortForInstance(networkClient, *bootstrapMachine.Status.InstanceID)
 	if err != nil {
 		return err
 	}
 
-	network, err := getNetworkByName(networkClient, cluster.Spec.ExternalNetworkID)
+	network, err := networks.Get(networkClient, cluster.Status.ExternalNetwork.ID).Extract()
```

Comment:
Do we need to fetch the network again given we already the `cluster.Status.ExternalNetwork.ID`?

Reply:
good catch! That makes no sense. Thank you!